lms/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -0,0 +1,142 @@
+"""
+Tests for Blocks Views
+"""
+
+import json
+
+import ddt
+from django.test import TestCase, RequestFactory
+from unittest import skip
+
+from student.tests.factories import UserFactory
+
+from .. import adapters
+from .. import views
+
+
+@ddt.ddt
+class TestAccessTokenView(TestCase):
+    """
+    Test class for AccessTokenView
+    """
+
+    dop_adapter = adapters.DOPAdapter()
+    dot_adapter = adapters.DOTAdapter()
+
+    def setUp(self):
+        super(TestAccessTokenView, self).setUp()
+        self.user = UserFactory()
+        self.dot_app = self.dot_adapter.create_public_client(user=self.user, client_id='dot-app-client-id')
+        self.dop_client = self.dop_adapter.create_public_client(user=self.user, client_id='dop-app-client-id')
+
+    def _post_access_token_request(self, user, client):
+        """
+        Make an HTTP POST request a token for the given user and client.
+
+        Returns an HTTP Response.
+        """
+        token_view = views.AccessTokenView.as_view()
+        request = RequestFactory().post('/', {
+            'client_id': client.client_id,
+            'grant_type': 'password',
+            'username': user.username,
+            'password': 'test',
+        })
+        return token_view(request)
```

Comment:
Question: Is there a reason this is tested by creating a Request object instead of calling django `reverse`?